File: server/datasets/python_source-code/Multiple-G1/renamed_39_bst.py
class Node_3:
    def __init__(self, val_3):
        self.val_3 = val_3
        self.lhs_3 = None
        self.rhs_3 = None

class Tree_3:
    def __init__(self):
        self.rt_3 = None

    def ins_3(self, val_3):
        if self.rt_3 is None:
            self.rt_3 = Node_3(val_3)
        else:
            self._ins_3_rec(self.rt_3, val_3)

    def _ins_3_rec(self, nd_3, val_3):
        if val_3 < nd_3.val_3:
            if nd_3.lhs_3 is None:
                nd_3.lhs_3 = Node_3(val_3)
            else:
                self._ins_3_rec(nd_3.lhs_3, val_3)
        elif val_3 > nd_3.val_3:
            if nd_3.rhs_3 is None:
                nd_3.rhs_3 = Node_3(val_3)
            else:
                self._ins_3_rec(nd_3.rhs_3, val_3)

    def srch_3(self, val_3):
        return self._srch_3_rec(self.rt_3, val_3)

    def _srch_3_rec(self, nd_3, val_3):
        if nd_3 is None or nd_3.val_3 == val_3:
            return nd_3
        if val_3 < nd_3.val_3:
            return self._srch_3_rec(nd_3.lhs_3, val_3)
        return self._srch_3_rec(nd_3.rhs_3, val_3)

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_g1/renamed_39_bst.java
class Node_3 {
    int val_3;
    Node_3 lhs_3, rhs_3;
    public Node_3(int item) {
        val_3 = item;
        lhs_3 = rhs_3 = null;
    }
}
public class Tree_3 {
    public static void main(String[] args) {
        Tree_3 bst = new Tree_3();
        bst.ins_3(50);
        bst.srch_3(50);
    }
    Node_3 rt_3;
    public void ins_3(int val_3) {
        rt_3 = in_rec_3(rt_3, val_3);
    }
    Node_3 in_rec_3(Node_3 rt_3, int val_3) {
        if (rt_3 == null) return new Node_3(val_3);
        if (val_3 < rt_3.val_3) rt_3.lhs_3 = in_rec_3(rt_3.lhs_3, val_3);
        else if (val_3 > rt_3.val_3) rt_3.rhs_3 = in_rec_3(rt_3.rhs_3, val_3);
        return rt_3;
    }
    Node_3 srch_3(int val_3) {
        return sr_rec_3(rt_3, val_3);
    }
    Node_3 sr_rec_3(Node_3 rt_3, int val_3) {
        if (rt_3 == null || rt_3.val_3 == val_3) return rt_3;
        if (rt_3.val_3 < val_3) return sr_rec_3(rt_3.rhs_3, val_3);
        return sr_rec_3(rt_3.lhs_3, val_3);
    }
}